sdk/core/azure-core/src/main/java/com/azure/core/http/policy/BearerTokenAuthenticationPolicy.java

```diff
@@ -75,7 +78,7 @@ public Mono<Void> authorizeRequest(HttpPipelineCallContext context) {
         if (this.scopes == null) {
             return Mono.empty();
         }
-        return setAuthorizationHeaderHelper(context, new TokenRequestContext().addScopes(this.scopes), false);
+        return setAuthorizationHeaderHelper(context, new TokenRequestContext().addScopes(this.scopes).setCaeEnabled(true), false);
```

Comment:
Is there a CHANGELOG entry we should add discussing that CAE challenge is enabled by default? Also, is there a way to disable it by default in case a bug scenario is hit?

Reply:
There appears to be a small difference from `ArmChallengeAuthenticationPolicy` used for ARM CAE.
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-management/src/main/java/com/azure/core/management/http/policy/ArmChallengeAuthenticationPolicy.java#L59

Though I assume they are handling for the same CAE challange.

What would happen to `ArmChallengeAuthenticationPolicy` after this PR merged?
Should mgmt lib switch to use `BearerTokenAuthenticationPolicy` directly (and deprecate `ArmChallengeAuthenticationPolicy`)?

Reply:
I'll let @billwert comment when he returns, but I believe we should be able to replace `ArmChallengeAuthenticationPolicy` with `BearerTokenAuthenticationPolicy`. It is possible that the older policy is based on a slightly different format of the challenge.

Reply:
Yes, I think this be the plan too. New SDK generated would use `BearerTokenAuthenticationPolicy` directly, while `ArmChallengeAuthenticationPolicy` would still be in place (maybe updated) for backward-compatibility.

One thing we need to verify is that `ArmChallengeAuthenticationPolicy` would still work after the change to `BearerTokenAuthenticationPolicy` (its base class).

Reply:
I believe that the older implementation of CAE that `ArmChallengeAuthenticationPolicy` targeted was never actually implemented. Does it perform any other function besides handling CAE challenges? If not, there may be no reason to keep it for backwards compat.

Reply:
👍 ARM CAE never left preview so there's no need to keep the client code we wrote for it years ago

Reply:
> I believe that the older implementation of CAE that `ArmChallengeAuthenticationPolicy` targeted was never actually implemented. Does it perform any other function besides handling CAE challenges? If not, there may be no reason to keep it for backwards compat.

We actually verified the behavior of CAE in `ArmChallengeAuthenticationPolicy` via end2end test on a prod tenant in 2021 on ARM Microsoft.Resources RP. So, customer can use that since 2021; though I don't know how many customer indeed depends on this preview CAE feature.

Reply:
ARM disabled their CAE implementation early this year. It had no users at that time

Reply:
Thanks for the discussion while I was gone!

It sounds like the conclusion here is that we should remove `ArmChallengeAuthenticationPolicy` and use what we are adding here. @weidongxu-microsoft, do you want to do that in a follow up once this is in?